### DISCUSSION TITLE
Displaying glyphs that do not fit in glyph cell in font overview

### DISCUSSION THREAD

**Discussion by khaledhosny** [fontra/fontra#1890] [Ideas & Feature Requests]
Since you are working on font overview, I thought to share this idea with you. I work on fonts that often have too tall or too wide glyphs that do not fit in the glyph cell (for example math fonts that have delimiters or big operator growing in side, or Arabic fonts with too wide glyphs like ڪ), and font editors will just clip the glyph, which makes finding these glyphs harder to find by eye.

I was wondering if it would be possible to allow glyph cell to span multiple columns, rows, or both if the glyph desn’t fit in the cell, something like this?
<img width="778" alt="image" src="https://github.com/user-attachments/assets/c644f1ab-28b3-4f62-bc11-4f0bf0be610f" />

I haven’t done any GUI development in a long time, so I don’t know how much of this is technically feasible, but also does it sound like a good idea design wise?

**Comment by justvanrossum** [comment]
The cells will be variable width (currently based on the advance width), so it won't be a true grid. But to be honest, I haven't thought yet about super tall glyphs, and I'm not sure how that will fit in the current design. I'll try to keep it in mind.

**Reply by khaledhosny** [reply]
I gave fontra’s fontoverview a try. The cells has a minimal width? Because too narrow glyphs dob’t get narrow cells. Wide glyphs are indeed fully visible, which is a big improvement!

I feel that it would have visually looked better if glyph width was multiples of that minimal cell width, since it currently looks a bit jarring. Here is how a portion of Noto Sans Math looks:
![image](https://github.com/user-attachments/assets/02379753-7a95-4394-9f6b-34ee92df7e4e)

**Reply by justvanrossum** [reply]
Thanks for trying!

Yeah, there's a minimal width, so for narrow-ish fonts it becomes a grid view. I don't mind the jagginess that much, but I'll consider keeping a horizontal grid (perhaps half if the current minimum width).
